File: backend/src/main/java/ru/otus/backend/db/entity/Status.java
package ru.otus.backend.db.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

@Data
@Table("statuses")
@AllArgsConstructor
@NoArgsConstructor
public class Status {
    @Id
    private Long id;
    private String name;

    public Status(String name) {
        this.name = name;
    }
}

File: frontend/src/app/main-module/models/status.ts
export class Status {

    id: string;
    name: string;

    static cloneStatus(status: Status): Status {
        if (!status) {
            return null;
        }

        let cloneStatus: Status = new Status();
        cloneStatus.id = status.id;
        cloneStatus.name = status.name;
        return cloneStatus;
    }
}
